core/src/main/java/apoc/util/JsonUtil.java

```diff
@@ -78,20 +85,30 @@ private static Configuration getJsonPathConfig(List<String> options) {
         }
     }
     
-    public static Stream<Object> loadJson(String url, Map<String,Object> headers, String payload) {
-        return loadJson(url,headers,payload,"", true, null, null);
+    public static Stream<Object> loadJson(String url, Map<String,Object> headers, String payload, ApocConfig apocConfig) {
+        return loadJson(url,headers,payload,"", true, null, null, apocConfig);
     }
     
-    public static Stream<Object> loadJson(Object urlOrBinary, Map<String,Object> headers, String payload, String path, boolean failOnError, List<String> options) {
-        return loadJson(urlOrBinary, headers, payload, path, failOnError, null, options);
+    public static Stream<Object> loadJson(Object urlOrBinary, Map<String,Object> headers, String payload, String path, boolean failOnError, List<String> options, ApocConfig apocConfig) {
+        return loadJson(urlOrBinary, headers, payload, path, failOnError, null, options, apocConfig);
     }
     
-    public static Stream<Object> loadJson(Object urlOrBinary, Map<String,Object> headers, String payload, String path, boolean failOnError, String compressionAlgo, List<String> options) {
+    public static Stream<Object> loadJson(Object urlOrBinary, Map<String,Object> headers, String payload, String path, boolean failOnError, String compressionAlgo, List<String> options, ApocConfig apocConfig) {
         try {
             if (urlOrBinary instanceof String) {
                 String url = (String) urlOrBinary;
```

Comment:
Shouldn't this check be done in `checkReadAllowed`, where the other access rules as processed? With the check here, you can still make calls that violate the blocklist using `apoc.load.xml`, `apoc.xml.import`, `apoc.load.xmlSimple`, `apoc.load.xls`, `apoc.cypher.runFiles` and so on. 

Similarly, we should process the blocklist in `checkWriteAllowed`, otherwise people can violate the blocklist by sending POST requests and other output network connections?

Reply:
yes we should not pass the full apoc config through, either just what we need or better like Jake said handle it in the single place where we control the loading.

Reply:
https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/4.3/core/src/main/java/apoc/util/FileUtils.java#L142-L241

Reply:
Also you don't need to pass `apocConfig` through but use `ApocConfig.getConfig().methodCall()`

there are already some methods for checking, we can handle the address as part of that.
so you would rather do something like `ApocConfig#isIpAllowed(ip)` or `ApocConfig#isHostNameAllowed(hostname)`
need to make sure though to resolve the final hostname properly to ip-address (after all the http-redirects are handled).

https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/4.3/core/src/main/java/apoc/ApocConfig.java#L250-L262

Reply:
@ncordon please look at the comments above

Reply:
@jexp I have addressed the concern for redirects in a pending PR: https://github.com/neo-technology/neo4j/pull/14171. The logic for the blocking lives in the kernel of the database and here I am reusing the same logic. So once that's merged we should be safe with respect to redirects (I hope).

@conker84 I get the need for `checkReadAllowed`, but `checkWriteAllowed` works only with filenames, which this blocklist doesn't include, judging by its signature?

Do we have any situation where we can do a POST to a path in a remote server?

Reply:
This has been addressed. 

With respect to `checkWriteAllowed`, we've deemed that change is not necessary as it only affects files. POST requests (load json ones for example) go through the `checkReadAllowed`.

Reply:
write list also works with urls e.g. for s3 URLs

Reply:
`checkWriteAllowed` is only used with things that require a physical file I believe:
<img width="881" alt="checkWriteAllowed" src="https://user-images.githubusercontent.com/5649971/152525216-a9107cb1-5a6b-41ab-9197-8ea9a9f0b309.png">